pipeline/extract.py

```diff
@@ -0,0 +1,126 @@
+
+def fetch_file_content(s3: client, file_name: str, topic: list[str]) -> dict:
+    """Checks files from S3 fir keywords and returns relevant data if keyword is found"""
+    try:
+        file_obj = s3.get_object(Bucket=os.environ.get("S3_BUCKET_NAME"), Key=file_name)
+        file_content = file_obj['Body'].read().decode('utf-8')
+        for keyword in topic:
+            if keyword in file_content:
+                logging.info("Keyword '%s' found in %s", keyword, file_name)
+                return {
+                    "Text": file_content,
+                    "Keyword": keyword,
+                }
+    except FileNotFoundError as e:
+        logging.error("No files found in S3: %s", e)
+    return None
+
+def access_bluesky_files(s3: client, topic: list[str]) -> list[str]:
+    """Accesses files from S3 and returns a list of texts with the topic present"""
+    continuation_token = None
+    file_names = []
+
+    #continously fetches .txt files from bucket until all files have been fetched
+    while True:
+        bucket_parameters = {'Bucket': os.environ.get("S3_BUCKET_NAME")}
+        if continuation_token:
+            bucket_parameters['ContinuationToken'] = continuation_token
+
+        response = s3.list_objects_v2(**bucket_parameters)
+        contents = response.get("Contents")
+
+        for file in contents:
+            if file['Key'].endswith(".txt"):
+                file_names.append(file['Key'])
+
+        continuation_token = response.get('NextContinuationToken')
+        if not continuation_token:
+            break
+
+    matching_texts = []
+    with ThreadPoolExecutor(max_workers=40) as thread_pool:
+        submitted_tasks = [thread_pool.submit(fetch_file_content, s3,
+                                   file_name, topic) for file_name in file_names]
+
+        for completed_task in as_completed(submitted_tasks):
+            matching_result = completed_task.result()
+            if matching_result:
+                matching_texts.append(matching_result)
+
+    return matching_texts
+
+def create_dataframe(topic: list[str]) -> pd.DataFrame:
+    """Main function to extract data and return a DataFrame of the relevant data"""
+    s3 = s3_connection()
+
+    results = access_bluesky_files(s3, topic)
+    logging.info(type(results))
+
+    return pd.DataFrame(results)
+
+def fetch_suggestions(pytrend: TrendReq, keyword: str):
+    """Fetch and print suggestions for a given keyword."""
+    return pytrend.suggestions(keyword=keyword)
+
+def main(topic: list[str]) -> pd.DataFrame:
+    """Extracts data from S3 Bucket and google trends"""
+    extract_dataframe = create_dataframe(topic)
+
+    pytrend = initialize_trend_request()
+    extract_dataframe['related_terms'] = ""
+
+    for keyword in topic:
+        extract_dataframe.loc[extract_dataframe['Keyword']
+                              == keyword, 'related_terms'] = ",".join(
+            [suggestion['title'] for suggestion in fetch_suggestions(pytrend, keyword)]
```

Comment:
Could we use a lambda function here instead of a for loop:
```
extracted_df['related_terms'] = extracted_df['Keyword'].apply(
        lambda kw: ",".join(fetch_suggestions(pytrend, kw))
    )
```